### PR TITLE
 Code Modernization: Use `str_contains()`, `str_starts_with()` and `str_ends_with()`

### PR DESCRIPTION
**Pull request by xxsimoxx**
This is a backport of multiple WP changesets that changes the use of `str_pos` with `str_contains()`, `str_starts_with()` and `str_ends_with()`.

Those function comes from PHP 8 but polyfills are already defined in core.

**This PR is a work in progress as I have to check if other similar changeset can be backported.**

## Changesets

- WP-r55703: Code Modernization: Replace usage of `strpos()` with `str_starts_with()`.
- WP-r55988: Code Modernization: Replace usage of `strpos()` with `str_contains()`.
- WP-r55990: Code Modernization: Replace usage of `substr()` with `str_starts_with()` and `str_ends_with()`.
- WP-r56014: Code Modernization: Use `str_ends_with()` in a few more places.
- WP-r56019: Code Modernization: Use `str_starts_with()` and `str_ends_with()` in a few more places.
- WP-r56020: Code Modernization: Use `str_starts_with()` and `str_ends_with()` in a few more places.
- WP-r56021: Code Modernization: Use `str_contains()` in a few more places.
- WP-r56031: Code Modernization: Use `str_contains()` in a few more places.
- WP-r56032: Code Modernization: Use `str_contains()` in a few more places.
- WP-r56088: Revert use of `str_starts_with()` and `str_contains()` in `update-core.php`.
- WP-r56245: Code Modernization: Use `str_contains()` in a few more places.
- WP-r56417: Revert the last use of `str_starts_with()` in `update-core.php`.

## How has this been tested?
- Unit tests passing
- Local tests

**Notice: when merged we have to test migrating from 1.7 to a nightly.**

### CODE REVIEW

**Comment by mattyrob** [comment]
Great work so far. It looks like `str_contains()` polyfill is missing for PHP 7.4 and there are a couple of PHP coding standards issues.

**Comment by xxsimoxx** [comment]
Polyfills are in place but I've to check if is better backporting `compat.php` load order [from WP](https://github.com/WordPress/wordpress-develop/commit/3349f1a527761eed3aa2d48050d80a9b62fc6860) or do it our way.

**Comment by xxsimoxx** [comment]
Changed CP load sequence to get polyfills earlier. Taken from [this changeset](https://core.trac.wordpress.org/changeset/56241) that was partially reverted in [this commit](https://github.com/ClassicPress/ClassicPress/pull/1320/commits/9d277235442eef426ff44a3338f61646ee64f56c).

**This needs to be carefully reviewed.**

**Comment by xxsimoxx** [comment]
Other backports added:

- WP-r57126: Code Modernization: Use `str_starts_with()` in `WP_REST_Server::serve_request()`.
- WP-r55710: General: Restore `strpos()` check in `wp-admin/load-styles.php`.
- WP-r55703: Code Modernization: Replace usage of `strpos()` with `str_starts_with()`.
- WP-r52326: KSES: Use the polyfilled PHP 8 string functions in `_wp_kses_allow_pdf_objects()`

A `grep` for `strpos` on all files in `src` directory returned a list of files that I've checked against WordPress develop.
**Seems that there are no remaining changesets to backport.**

In [this commit](https://github.com/ClassicPress/ClassicPress/pull/1320/commits/f12fa9fbac5656898e960be7a4083a4f4befc5d4) I've manually changed those files because they belong to our codebase or our codebase has diverged from WP too much to get a future backport:

- `src/wp-admin/media.php`
- `src/wp-includes/classicpress/class-cp-customization.php`
- `src/wp-includes/classicpress/class-cp-debug-compat.php`
- `src/wp-includes/classicpress/class-wp-compat.php`
- `src/wp-includes/post.php`

File: src/wp-admin/includes/update-core.php [hunk 1]
```diff
@@ -602,9 +617,13 @@ function update_core( $from, $to ) {
...
 		}
-
+		/*
+		 * Note: str_starts_with() is not used here, as this file is included
+		 * when updating from older WordPress versions, in which case
+		 * the polyfills from wp-includes/compat.php may not be available.
+		 */
 		// Check if the language directory exists first.
-		if ( ! @is_dir( $lang_dir ) && 0 === strpos( $lang_dir, ABSPATH ) ) {
+		if ( ! @is_dir( $lang_dir ) && str_starts_with( $lang_dir, ABSPATH ) ) {
```
Comment (mattyrob) [inline src/wp-admin/includes/update-core.php:626]: As per the docblock above I think this ma need reverting to `0 === strpos()`

**Review by mattyrob — CHANGES_REQUESTED** [review]

File: src/wp-admin/includes/update-core.php (hunk 1 above)
Comment (xxsimoxx) [reply src/wp-admin/includes/update-core.php]: Reverted. Thanks.

**Review by mattyrob — APPROVED** [review]

**Pull request merged**